iaso/api/form_versions.py

```diff
@@ -132,6 +133,18 @@ def update(self, form_version, validated_data):
         return form_version
 
 
+class HasFormVersionPermission(HasFormPermission):
+    def has_object_permission(self, request, view, obj) -> bool:
+        if not self.has_permission(request, view):
+            return False
+
+        ok_forms = Form.objects_include_deleted.filter_for_user_and_app_id(
+            request.user, request.query_params.get("app_id")
+        )
+
+        return obj.form in ok_forms
```

Comment:
do not use `in` to check I made this mistake before somehow it make django fetch all the records. Use .filter().exists()

Reply:
I was actually thinking about it but the current code in [/api/forms.py](https://github.com/BLSQ/iaso/blob/e89da17f8e8cb2e0a522251a300ed51c14fa3726/iaso/api/forms.py#L30) has the same 

```python
return obj in Form.objects_include_deleted.filter_for_user_and_app_id(
    request.user, request.query_params.get("app_id")
)
```

So this should be corrected too then ?

Reply:
yep

Reply:
I corrected both and pushed it